crates/storage/libmdbx-rs/src/txn_manager.rs

```diff
@@ -73,34 +70,12 @@ impl TxnManager {
                                 )
                             })
                             .map(|_| TxnPtr(txn));
-
-                            #[cfg(feature = "read-tx-timeouts")]
-                            {
-                                use crate::transaction::TransactionKind;
-
-                                if res.is_ok() && flags == crate::transaction::RO::OPEN_FLAGS {
-                                    if let Some(read_transactions) = &read_transactions {
-                                        read_transactions.add_active(txn);
-                                    }
-                                }
-                            }
-
                             sender.send(res).unwrap();
                         }
                         TxnManagerMessage::Abort { tx, sender } => {
-                            #[cfg(feature = "read-tx-timeouts")]
-                            if let Some(read_transactions) = &read_transactions {
-                                read_transactions.remove_active(tx.0);
-                            }
```

Comment:
why is this not needed anymore? seems like `TxnManagerMessage` needs to enforce it is only sent write transactions with a new type `WriteTxPtr(TransactionPtr)` inside the message if we remove this, is this the plan?

Reply:
> seems like `TxnManagerMessage` needs to enforce it is only sent write

agreed

> with a new type `WriteTxPtr(TransactionPtr)`

unsure, I'd rather make `TxnManagerMessage` work with read-write transactions **only**, because we spawn all read transactions without it anyway. And do it in a separate PR.

Reply:
I removed it because we don't have an instance of `TransactionPtr` at the moment when we process the `TxnManagerMessage::Begin`. It's created later after we send the resulting pointer on the channel.

Reply:
alright, I would have added todo in code then for other devs to avoid confusion